lib/router/transition.js

```diff
@@ -58,7 +70,7 @@ function Transition(router, intent, state, error) {
       this.pivotHandler = handlerInfo.handler;
     }
 
-    this.sequence = Transition.currentSequence++;
+    this.sequence = router.currentSequence++;
     this.promise = state.resolve(checkForAbort, this)['catch'](function(result) {
```

Comment:
That function is lexically scoped in a place where it can reach the previous transition which means it can't be cleaned up until that property is reset.

Yes, we need to refactor it such that the definition of this section happens outside of the constructor. This will have the bonus win of being more performant.